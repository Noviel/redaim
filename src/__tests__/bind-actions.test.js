import {
  createAction,
  bindActions,
  bindActionsR
} from '../actions';

describe('bindActions', () => {
  it('should bind trigger to actions', () => {
    const actionsObject = {
      exec1: { exec: 'EXEC1', payload: { value: 1 } },
      exec2: { exec: 'EXEC2', payload: { value: 2 } },
      exec3: { exec: 'EXEC3', payload: { value: 3 } }
    };
    const boundActionsCreator = bindActions(createAction('MY_TRIGGER'))('MY_TARGET');
    const { exec1, exec2, exec3 } = boundActionsCreator(actionsObject);

    const result = {
      exec1: exec1().type,
      exec2: exec2().type,
      exec3: exec3().type
    };

    expect(result).toMatchSnapshot();
  });

  it('should bind target to actions', () => {
    const actionsObject = {
      exec1: { exec: 'EXEC1', payload: { value: 1 } },
      exec2: { exec: 'EXEC2', payload: { value: 2 } },
      exec3: { exec: 'EXEC3', payload: { value: 3 } }
    };
    const boundActionsCreator = bindActions(createAction('MY_TRIGGER'))('MY_TARGET');
    const { exec1, exec2, exec3 } = boundActionsCreator(actionsObject);

    const result = {
      exec1: exec1().meta.target,
      exec2: exec2().meta.target,
      exec3: exec3().meta.target
    };

    expect(result).toMatchSnapshot();
  });
});

describe('bindActionsR', () => {
  it('should bind trigger to actions', () => {
    const actionsObject = {
      exec1: { exec: 'EXEC1', payload: { value: 1 } },
      exec2: { exec: 'EXEC2', payload: { value: 2 } },
      exec3: { exec: 'EXEC3', payload: { value: 3 } }
    };
    const boundActionsCreator = bindActionsR(actionsObject);
    const { exec1, exec2, exec3 } = boundActionsCreator('TRIGGER')('special_target');

    const result = {
      exec1: exec1().type,
      exec2: exec2().type,
      exec3: exec3().type,
    };

    expect(result).toMatchSnapshot();
  });

  it('should bind target to actions', () => {
    const actionsObject = {
      exec1: { exec: 'EXEC1', payload: { value: 1 } },
      exec2: { exec: 'EXEC2', payload: { value: 2 } },
      exec3: { exec: 'EXEC3', payload: { value: 3 } }
    };
    const boundActionsCreator = bindActionsR(actionsObject);
    const { exec1, exec2, exec3 } = boundActionsCreator('TRIGGER')('special_target');

    const result = {
      exec1: exec1().meta.target,
      exec2: exec2().meta.target,
      exec3: exec3().meta.target
    };

    expect(result).toMatchSnapshot();
  });
});
