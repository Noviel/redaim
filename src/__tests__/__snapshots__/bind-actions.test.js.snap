// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bindActions should bind target to actions 1`] = `
Object {
  "exec1": "MY_TARGET",
  "exec2": "MY_TARGET",
  "exec3": "MY_TARGET",
}
`;

exports[`bindActions should bind trigger to actions 1`] = `
Object {
  "exec1": "MY_TRIGGER",
  "exec2": "MY_TRIGGER",
  "exec3": "MY_TRIGGER",
}
`;

exports[`bindActionsR should bind target to actions 1`] = `
Object {
  "exec1": "special_target",
  "exec2": "special_target",
  "exec3": "special_target",
}
`;

exports[`bindActionsR should bind trigger to actions 1`] = `
Object {
  "exec1": "TRIGGER",
  "exec2": "TRIGGER",
  "exec3": "TRIGGER",
}
`;
